src/embeddings.jl

```diff
@@ -53,22 +78,71 @@ Optimize an embedding by minimizing the fuzzy set cross entropy between the high
 # Arguments
 - `graph`: a sparse matrix of shape (n_samples, n_samples)
 - `embedding`: a vector of length (n_samples,) of vectors representing the embedded data points
-- `n_epochs`: the number of training epochs for optimization
+- `n_epochs`::Integer: the number of training epochs for optimization
 - `initial_alpha`: the initial learning rate
 - `gamma`: the repulsive strength of negative samples
 - `neg_sample_rate::Integer`: the number of negative samples per positive sample
 """
 function optimize_embedding(graph,
-                            embedding,
-                            n_epochs,
+                            embedding::AbstractVector{<:AbstractVector{<:AbstractFloat}},
```

Comment:
So one issue was having the `optimize_embedding` methods with both one embedding argument and with two (for query and ref), the call would not be properly dispatched because of the default arguments to `_a` and `_b`, and I was resolving this with type constraints. But I can remove the default arguments or remove the overloaded methods, if you have a preference towards either of those?

Reply:
If you have no preference, I will just remove the original `optimize_embedding` method so all calls will have to pass two embeddings (or the same one twice), I'm starting to think the overloading is more unnecessary/messy than good.